packages/core/src/utils.ts

```diff
@@ -131,6 +122,30 @@ export class Utils {
     const version = cid.version
     await ipfsApi.block.put(block, { format, mhtype, version, signal })
   }
+
+  /**
+   * Attempts to load CACAO capability from IPFS if present in protected header of commit
+   * @param commit the commit to load the capability for
+   * @param dispatcher instance of dispatcher to load from IPFS
+   * @returns a Cacao capability object if found, else null
+   */
+  static async extractCapability(commit: any, dispatcher: Dispatcher): Promise<any | null> {
```

Comment:
```suggestion
  static async extractCapability(commit: any, dispatcher: Dispatcher): Promise<CACAO | undefined> {
```

We should convert the type to CACAO before we return. If nothing is returned the function will return undefined.